tests/007-datagrams.cpp

```diff
@@ -279,15 +298,317 @@ namespace oxen::quic::test
+
+                        tls_promise.set_value(true);
+                        return 0;
+                    };
+
+            dgram_data_callback recv_dgram_cb = [&](bstring) {
+                log::debug(log_cat, "Calling endpoint receive datagram callback... data received...");
+
+                try
+                {
+                    data_counter += 1;
+                    data_promises.at(index).set_value(true);
+                    index += 1;
+                }
+                catch (std::exception& e)
+                {
+                    throw std::runtime_error(e.what());
+                }
+            };
+
+            opt::enable_datagrams split_dgram{Splitting::ACTIVE};
+
+            opt::local_addr server_local{};
+            opt::local_addr client_local{};
+
+            auto server_tls = GNUTLSCreds::make("./serverkey.pem"s, "./servercert.pem"s, "./clientcert.pem"s);
+            auto client_tls = GNUTLSCreds::make("./clientkey.pem"s, "./clientcert.pem"s, "./servercert.pem"s);
+            client_tls->set_client_tls_policy(outbound_tls_cb);
+
+            auto server_endpoint = test_net.endpoint(server_local, split_dgram, recv_dgram_cb);
+            REQUIRE_NOTHROW(server_endpoint->listen(server_tls));
+
+            opt::remote_addr client_remote{"127.0.0.1"s, server_endpoint->local().port()};
+
+            auto client = test_net.endpoint(client_local, split_dgram);
+            auto conn_interface = client->connect(client_remote, client_tls);
+
+            REQUIRE(tls_future.get());
+
+            REQUIRE(server_endpoint->datagrams_enabled());
+            REQUIRE(client->datagrams_enabled());
+
+            REQUIRE(conn_interface->datagrams_enabled());
+            REQUIRE(conn_interface->packet_splitting_enabled());
+
+            auto max_size = conn_interface->get_max_datagram_size();
+
+            std::basic_string<uint8_t> big_msg{}, small_msg{};
+            uint8_t v{0};
+
+            while (big_msg.size() < max_size)
+                big_msg += v++;
+
+            while (small_msg.size() < 500)
+                small_msg += v++;
+
+            conn_interface->send_datagram(std::basic_string_view<uint8_t>{big_msg});
+            conn_interface->send_datagram(std::basic_string_view<uint8_t>{big_msg});
+            conn_interface->send_datagram(std::basic_string_view<uint8_t>{small_msg});
+            conn_interface->send_datagram(std::basic_string_view<uint8_t>{big_msg});
+            conn_interface->send_datagram(std::basic_string_view<uint8_t>{small_msg});
+
+            for (auto& f : data_futures)
+                REQUIRE(f.get());
+
+            REQUIRE(data_counter == int(n));
 
             test_net.close();
         };
     };
+
+    TEST_CASE("007 - Datagram support: Rotating Buffer, Induced Loss", "[007][datagrams][execute][split][rotating][loss]")
+    {
+        if (disable_rotating_buffer)
+            SKIP("Rotating buffer testing not enabled for this test iteration!");
+
```

Comment:
```suggestion

#ifdef NDEBUG
        SKIP("Induced loss test requires a debug build");
#else
```

Reply:
and `#endif` down at the bottom of the section

Reply:
added!